packages/video_player/video_player/android/src/main/java/io/flutter/plugins/videoplayer/Messages.java

```diff
@@ -0,0 +1,338 @@
+// Autogenerated from Pigeon (v0.1.0-experimental.3), do not edit directly.
+// See also: https://pub.dev/packages/pigeon
+
+package io.flutter.plugins.videoplayer;
+
+import java.util.HashMap;
+
+import io.flutter.plugin.common.BasicMessageChannel;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugin.common.StandardMessageCodec;
+
+/** Generated class from Pigeon. */
+public class Messages {
+
+  /** Generated class from Pigeon that represents data sent in messages. */
+  public static class VoidMessage {
+    HashMap toMap() {
+      HashMap<String, Object> toMapResult = new HashMap<String, Object>();
+      return toMapResult;
+    }
+    static VoidMessage fromMap(HashMap map) {
+      VoidMessage fromMapResult = new VoidMessage();
+      return fromMapResult;
+    }
+  }
+
+  /** Generated class from Pigeon that represents data sent in messages. */
+  public static class TextureMessage {
+    private Long textureId;
+    public Long getTextureId() { return textureId; }
+    public void setTextureId(Long setterArg) { this.textureId = setterArg; }
+
+    HashMap toMap() {
+      HashMap<String, Object> toMapResult = new HashMap<String, Object>();
+      toMapResult.put("textureId", textureId);
+      return toMapResult;
+    }
+    static TextureMessage fromMap(HashMap map) {
+      TextureMessage fromMapResult = new TextureMessage();
+      fromMapResult.textureId = (int)map.get("textureId");
+      return fromMapResult;
+    }
+  }
+
+  /** Generated class from Pigeon that represents data sent in messages. */
+  public static class CreateMessage {
+    private String asset;
+    public String getAsset() { return asset; }
+    public void setAsset(String setterArg) { this.asset = setterArg; }
+
+    private String uri;
+    public String getUri() { return uri; }
+    public void setUri(String setterArg) { this.uri = setterArg; }
+
+    private String package;
+    public String getPackage() { return package; }
+    public void setPackage(String setterArg) { this.package = setterArg; }
+
+    private String formatHint;
+    public String getFormatHint() { return formatHint; }
+    public void setFormatHint(String setterArg) { this.formatHint = setterArg; }
+
+    HashMap toMap() {
```

Comment:
I'm thinking, since the APIs are directional in the IDL, this should theoretically never be needed right? Could we omit this by default unless we annotate it somehow?